services/chat/api.go

```diff
@@ -244,14 +244,19 @@ func (api *API) getChannelGroups(ctx context.Context, channelGroupID string) (ma
 			}
 		}
 
-		for _, chat := range channels {
-			if chat.CommunityID == community.IDString() && chat.Active {
+		// Looping The community's chats first makes sure we only get the chats that are reallay part of the community
+		for communityChatID := range community.Chats() {
+			for _, chat := range channels {
+				if communityChatID != chat.CommunityChatID() || !chat.Active {
+					continue
```

Comment:
Could you please add a warning here to determine if that was indeed the issue? This would also let us understand better what happened. Once we have the chatID that is the culprit we could investigate the database.

Reply:
Added the log. Please check the code again since I switched back to the old loop

Reply:
Looks good :+1: